luigi/contrib/ftp.py

```diff
@@ -131,14 +131,11 @@ def _sftp_exists(self, path, mtime):
         return exists
 
     def _ftp_exists(self, path, mtime):
-        path_parts = path.split('/')
-        path = '/'.join(path_parts[:-1])
-        fn = path_parts[-1]
-
-        files = self.conn.nlst(path)
+        dirname = os.path.dirname(path)
+        files = self.conn.nlst(dirname)
 
         exists = False
-        if fn in files:
+        if path in files:
             if mtime:
```

Comment:
@daveFNbuck , I believe i used split and join over `os.path.dirname()` and `posixpath.basename` because I also needed the filename such that I could check for existence in the returned results from `nlst`.

This is where @andreabedini and I's FTP servers appear to be performing `nlst` differently. The servers I deal with only return filenames, whereas his returns fully qualified path names. I believe this is the main concern we need to code around.